src/services/videoconference/index.js

```diff
@@ -43,319 +43,314 @@ const { ObjectId } = require('../../helper/compare');
-			&& Object.values(SCOPE_NAMES).includes(params.scopeName);
-	}
+	// check permissions and set role
+	const scopePermissionService = app.service(`/${permissionScopeName}/:scopeId/userPermissions`);
+	const { [user.id]: userPermissionsInScope } = await scopePermissionService.find({
+		route: { scopeId: permissionScopeId },
+		query: { userId: user.id },
+	});
+
+	let logoutURL = `${CLIENT_HOST}/${permissionScopeName}/${permissionScopeId}`;
+	if (permissionScopeName === 'teams') logoutURL += '?activeTab=events';
+	if (permissionScopeName === 'courses') logoutURL += '?activeTab=tools';
+
+	return {
+		scopeTitle,
+		userPermissionsInScope,
+		logoutURL,
+	};
+}
 
-	static getHighestVideoconferencePermission(permissions) {
-		if (permissions.includes(PERMISSIONS.START_MEETING)) return PERMISSIONS.START_MEETING;
-		if (permissions.includes(PERMISSIONS.JOIN_MEETING)) return PERMISSIONS.JOIN_MEETING;
-		return null;
-	}
+function idAndScopeNameAreValid(params) {
+	// event ids are from postgres instead of mongo
+	const scopeIdMatchesEventId = (id) => /^[0-9a-f-]{36}$/.test(id);
+	return (ObjectId.isValid(params.scopeId) || scopeIdMatchesEventId(params.scopeId))
+		&& Object.values(SCOPE_NAMES).includes(params.scopeName);
+}
 
-	static createResponse(status, state, permissions, options = null, url) {
-		const permission = VideoconferenceBaseService
-			.getHighestVideoconferencePermission(permissions);
-		return {
-			status, state, permission, options, url,
-		};
-	}
+function getHighestVideoconferencePermission(permissions) {
+	if (permissions.includes(PERMISSIONS.START_MEETING)) return PERMISSIONS.START_MEETING;
+	if (permissions.includes(PERMISSIONS.JOIN_MEETING)) return PERMISSIONS.JOIN_MEETING;
+	return null;
+}
+
+function createResponse(status, state, permissions, options = null, url) {
+	const permission = getHighestVideoconferencePermission(permissions);
+	return {
+		status, state, permission, options, url,
+	};
+}
 
 
-	/**
- * fetchs the VideoconferenceModel with given scopeName and scopeId and returns it.
- * the model will be defined when a videoconference is created/starts.
+/**
+ * Fetches the VideoconferenceModel with given scopeName and scopeId and returns it.
+ * the model will be defined when a video conference is created/starts.
  * some of the options are reused from other users for join link generation
+ *
  * @param {String} scopeName
  * @param {String} scopeId
  * @re
  */
-	static async getVideocenceMetadata(scopeName, scopeId, returnAsObject = false) {
-		const modelDefaults = VideoconferenceBaseService.getDefaultModel(scopeName, scopeId);
-		const videoconferenceMetadata = await VideoconferenceModel
-			.findOne(modelDefaults).exec();
-		if (returnAsObject && videoconferenceMetadata !== null) {
-			return videoconferenceMetadata.toObject();
-		}
-		return videoconferenceMetadata;
+async function getVideocenceMetadata(scopeName, scopeId, returnAsObject = false) {
+	const modelDefaults = getDefaultModel(scopeName, scopeId);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['getDefaultModel' was used before it was defined.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5401970)